File: problems/[E]344.Reverse-String.kt
/**
 * Runtime: 232 ms, faster than 84.96% of Kotlin online submissions for Reverse String.
 * Memory Usage: 48 MB, less than 81.96% of Kotlin online submissions for Reverse String.
 */
class Solution {
    fun reverseString(s: CharArray): Unit {
        val size = s.size
        val lastIndex = s.lastIndex
        for (i in 0 until size / 2) {
            val temp = s[i]
            s[i] = s[lastIndex - i]
            s[lastIndex - i] = temp 
        }
    }
}

File: python/[E]344.Reverse-String.py
# Runtime: 196 ms, faster than 66.81% of Python3 online submissions for Reverse String.
# Memory Usage: 18.6 MB, less than 57.88% of Python3 online submissions for Reverse String.
class Solution:
    def reverseString(self, s: List[str]) -> None:
        lastIndex = len(s) - 1
        if lastIndex <= 0: return
        
        for i in range(0, int(lastIndex / 2) + 1):
            temp = s[i]
            s[i] = s[lastIndex - i]
            s[lastIndex - i] = temp
        
